julia-mode.el

```diff
@@ -551,24 +551,36 @@ the (possibly narrowed) buffer, so there is nowhere else to go."
 (defun julia-indent-hanging ()
   "Calculate indentation for lines that follow \"hanging\"
 operators (operators that end the previous line) as defined in
-`julia-hanging-operator-regexp'. An assignment operator ending
-the previous line increases the indent as do the other operators
-unless another operator is found two lines up. Previous line
-means previous line after skipping blank lines and lines with
-only comments."
-  (let (prev-indent)
+`julia-hanging-operator-regexp'. Previous line means previous
+ line after skipping blank lines and lines with only comments.
+The algorithm scans previous rows, keeping track of their
+indentation, until it reaches 0 indentation or has seen two
+hanging operators."
+  (let ((prev-indent) (prev-paren-indent) (hanging-match) (double-hanging) (best-indent))
     (save-excursion
       (when (> (julia-prev-line-skip-blank-or-comment) 0)
         (setq prev-indent (current-indentation))
-        (when (julia--hanging-operator-p)
-          (if (and (> (julia-prev-line-skip-blank-or-comment) 0)
-                   (julia--hanging-operator-p))
-              ;; two preceding hanging operators => indent same as line
-              ;; above
-              prev-indent
-            ;; one preceding hanging operator => increase indent from line
-            ;; above
-            (+ julia-indent-offset prev-indent)))))))
+        (setq prev-paren-indent (julia-paren-indent))
+        (setq hanging-match (julia--hanging-operator-p))
+        ;; Base case: one preceding hanging operator => increase indent from preceding line
+        (if hanging-match
+            (setq best-indent (+ (current-indentation) julia-indent-offset)))
+        ;; Loop rows back until we run out of indentation or hit a second hanging indent
+        (while (and hanging-match
+                    (not double-hanging)
+                    (> (current-indentation) 0)
+                    (> (julia-prev-line-skip-blank-or-comment) 0))
+          (setq double-hanging (julia--hanging-operator-p))
+          ;; Special case: two hanging operator indents already => use the first one
+          (if double-hanging
+              (setq best-indent prev-indent))
+          ;; Special case: open paren followed by one hanging operator
+          (if (and (not double-hanging)
+                   prev-paren-indent)
+              (setq best-indent (+ (current-indentation) julia-indent-offset)))
+          (setq prev-indent (current-indentation))
+          (setq prev-paren-indent (julia-paren-indent))))
+      best-indent)))
```

Comment:
When `(> (julia-prev-line-skip-blank-or-comment) 0)` is `nil`, I think this just returns `best-indent` which is `nil`. Would it make sense to just bail early, eg
```emacs-lisp
(when (> (julia-prev-line-skip-blank-or-comment) 0)
  (save-excursion
    (let* ((prev-indent (current-indentation))
           (prev-paren-indent (julia-paren-indent))
           (hanging-match (julia--hanging-operator-p))
           (double-hanging)
           (best-indent))
      ;; Base case: one preceding hanging operator => increase indent from preceding line
      (if hanging-match
          (setq best-indent (+ (current-indentation) julia-indent-offset)))
      ;; Loop rows back until we run out of indentation or hit a second hanging indent
      (while (and hanging-match
                  (not double-hanging)
                  (> (current-indentation) 0)
                  (> (julia-prev-line-skip-blank-or-comment) 0))
        (setq double-hanging (julia--hanging-operator-p))
        ;; Special case: two hanging operator indents already => use the first one
        (when double-hanging
          (setq best-indent prev-indent))
        ;; Special case: open paren followed by one hanging operator
        (when (and (not double-hanging) prev-paren-indent)
          (setq best-indent (+ (current-indentation) julia-indent-offset)))
        (setq prev-indent (current-indentation))
        (setq prev-paren-indent (julia-paren-indent)))
      best-indent)))
```
(note: **did not test this**, and this is just a style suggestion, feel free to ignore)

Reply:
Yeah, makes sense.

```julia
(when (> (julia-prev-line-skip-blank-or-comment) 0)
  (save-excursion
```
has to be 
```julia
  (save-excursion
     (when (> (julia-prev-line-skip-blank-or-comment) 0)
```
or calling the indentation moves `(point)` back by one row. Somehow it also seemed to get caught in an infinite loop or something.

Swapping the two rows works fine, tests pass. I've pushed a patch with these changes.